test/runtests.jl

```diff
@@ -150,6 +150,18 @@ struct IncompleteInterval <: AbstractInterval{Int} end
         @test isempty(ClosedInterval(B, A))
     end
 
+    @testset "isapprox" begin
+        @test 1..2 ≈ 1..2
+        @test 1..2 ≈ (1+1e-10)..2
+        @test 1..2 ≉ 1..2.01
+        @test 10..11 ≈ 10.1..10.9  rtol=0.01
+        @test 10..11 ≈ 10.1..10.9  atol=0.1
+        @test 10..11 ≉ 10.1..10.9  rtol=0.005
+        @test 10..11 ≉ 10.1..10.9  atol=0.05
+        @test 0..1 ≈ eps()..1
+        @test OpenInterval(0, 1) ≈ ClosedInterval(0, 1)
```

Comment:
There are two mathematical points of view for "isapprox":

* The measure of the symmetric difference between two intervals is approximately zero.
* The two real values that represent the interval are approximately equal.

`OpenInterval(0, 1) ≈ ClosedInterval(0, 1)` seems fine with the first point, but I think the second will be preferred because many users care about the precisions of the floating point numbers.

```julia
julia> i3 = Interval{:open,:closed}(1,2)
1..2 (open–closed)

julia> i4 = Interval{:closed,:open}(1,2)
1..2 (closed–open)

julia> isleftopen(i3)
true

julia> isleftopen(i4)
false

julia> i3 ≈ i4  # Should be false
true
```

Reply:
>Also, your suggestions in the other thread (eg `3..1 ≈ 5..1`) follow the first definition (sym diff) and not the second (endpoints are ≈).

:laughing:  Yes, the second point has exceptions with empty intervals.

>Anyway, to make it non-controversial, I only allow comparison of intervals with the same "closedness" now.

Thanks for the update! I still feel `i3 ≈ i4` should be `false`, but this can be changed in another PR.